lib/improver/utilities/save.py

```diff
@@ -173,6 +174,9 @@ def save_netcdf(cubelist, filename):
                   if key not in global_keys}
 
     cubelist = _append_metadata_cube(cubelist, global_keys)
-    iris.fileformats.netcdf.save(cubelist, filename, local_keys=local_keys,
+    # save atomically by writing to a temporary file and then renaming
+    ftmp = filename + '.tmp'
```

Comment:
Suggest explicitly converting to string here eg. `str(filename)` to avoid a TypeError if the filename has been passed in as a pathlib.Path object. Recent versions of Iris and netCDF4 do seem to be able to handle pathlib objects, so we should handle them safely.